test/framework/dockerfile_provider.go

```diff
@@ -18,41 +18,63 @@ import (
 	"fmt"
 	"io/ioutil"
 	"os"
-	"path/filepath"
 	"strings"
 
 	kogitores "github.com/kiegroup/kogito-cloud-operator/pkg/controller/kogitoapp/resource"
 )
 
+const (
+	quarkusJVMApplicationBinarySuffix    = "-runner.jar"
+	quarkusNativeApplicationBinarySuffix = "-runner"
+	springBootApplicationBinarySuffix    = ".jar"
+)
+
 // KogitoApplicationDockerfileProvider is the API to provide Dockerfile content for image creation based on built project content
 type KogitoApplicationDockerfileProvider interface {
 	// GetDockerfileContent returns Dockerfile content for image creation
-	GetDockerfileContent() string
+	GetDockerfileContent() (string, error)
 }
 
 type kogitoApplicationDockerfileProviderStruct struct {
 	projectLocation         string
 	imageName               string
 	applicationBinarySuffix string
 	libFolderNeeded         bool
+	matchApplicationBinary  func(filename string) (match bool)
 }
 
 var quarkusNonNativeKogitoApplicationDockerfileProvider = kogitoApplicationDockerfileProviderStruct{
 	imageName:               kogitores.KogitoQuarkusJVMUbi8Image,
-	applicationBinarySuffix: "-runner.jar",
+	applicationBinarySuffix: quarkusJVMApplicationBinarySuffix,
 	libFolderNeeded:         true,
+	matchApplicationBinary: func(filename string) (match bool) {
+		return strings.HasSuffix(filename, quarkusJVMApplicationBinarySuffix)
+	},
 }
 
 var quarkusNativeKogitoApplicationDockerfileProvider = kogitoApplicationDockerfileProviderStruct{
 	imageName:               kogitores.KogitoQuarkusUbi8Image,
-	applicationBinarySuffix: "-runner",
+	applicationBinarySuffix: quarkusNativeApplicationBinarySuffix,
 	libFolderNeeded:         false,
+	matchApplicationBinary: func(filename string) (match bool) {
+		return strings.HasSuffix(filename, quarkusNativeApplicationBinarySuffix)
+	},
 }
 
 var springbootKogitoApplicationDockerfileProvider = kogitoApplicationDockerfileProviderStruct{
 	imageName:               kogitores.KogitoSpringbootUbi8Image,
-	applicationBinarySuffix: ".jar",
+	applicationBinarySuffix: springBootApplicationBinarySuffix,
 	libFolderNeeded:         false,
+	matchApplicationBinary: func(filename string) (match bool) {
+		if strings.HasSuffix(filename, springBootApplicationBinarySuffix) {
+			// Need to exclude sources and tests
+			if strings.HasSuffix(filename, "-sources"+springBootApplicationBinarySuffix) || strings.HasSuffix(filename, "-tests"+springBootApplicationBinarySuffix) {
+				return false
+			}
+			return true
```

Comment:
I think it would be clearer using:

```go
return !strings.Contains(filename, "-sources.") && !strings.Contains(filename, "-tests.")
```

Or even:

```go
return strings.HasSuffix(filename, springBootApplicationBinarySuffix) && !strings.Contains(filename, "-sources.") && !strings.Contains(filename, "-tests.")
```

Reply:
Good point, applied.